contracts/axelar-gateway/tests/test.rs

```diff
@@ -490,3 +492,346 @@ fn upgrade_unauthorized() {
     assert_invoke_auth_err!(not_owner, client.try_upgrade(&new_wasm_hash));
     assert_invoke_auth_err!(client.operator(), client.try_upgrade(&new_wasm_hash));
 }
+
+#[test]
+#[should_panic]
+/// TODO: figure out how to detect error in constructor failure
+fn fail_initialization_with_empty_signer_set() {
+    let env = Env::default();
+    env.mock_all_auths();
+    let owner = Address::generate(&env);
+    let operator = Address::generate(&env);
+
+    let empty_signer_set = Vec::<WeightedSigners>::new(&env);
+    let domain_separator: BytesN<32> = BytesN::random(&env);
+    let previous_signers_retention = randint(0, 10) as u64;
+    let minimum_rotation_delay = 0;
+    let initial_signers = empty_signer_set;
+
+    // should panic because of empty signer set
+    env.register(
+        AxelarGateway,
+        (
+            owner,
+            operator,
+            domain_separator,
+            minimum_rotation_delay,
+            previous_signers_retention as u64,
+            initial_signers,
+        ),
+    );
+}
+
+#[test]
+#[should_panic(expected = "failed ED25519 verification")]
+fn fail_validate_proof_invalid_signatures() {
+    let (env, signers, client) = setup_env(randint(0, 10), randint(1, 10));
+
+    let (message, _) = generate_test_message(&env);
+    let messages = vec![&env, message.clone()];
+
+    let msg_hash: BytesN<32> = BytesN::random(&env);
+    let proof = generate_proof(&env, msg_hash, signers);
+
+    // should panic, proof is for different message hash
+    // NOTE: panic occurs in std function, cannot handle explicitly
+    client.approve_messages(&messages, &proof);
```

Comment:
instead of panicking, maybe test that the returned error when formatted contains the expected string (maybe extend `assert_contract_err` macro to support a rule that takes a string literal). Also, is it possible to check the typed error instead?
https://docs.rs/soroban-sdk/latest/soroban_sdk/testutils/ed25519/enum.Error.html
https://docs.rs/soroban-env-common/latest/soroban_env_common/struct.Error.html